lib/index-static-watcher.js

```diff
@@ -0,0 +1,120 @@
+// This is essentially a webpack-based compiler for
+// index-static.jsx. Much of the code is largely
+// based on James Long's "Backend Apps With Webpack"
+// series of blog posts:
+//
+// http://jlongster.com/Backend-Apps-with-Webpack--Part-I
+//
+// Doing this allows us to use the same build toolchain
+// for node and the browser, and also increases
+// responsiveness during development via hot-loading.
+
+var fs = require('fs');
+var path = require('path');
+var webpack = require('webpack');
+
+function IndexStaticWatcher(options) {
+  var self = {};
+  var outputDir = options.outputDir;
+  var outputFilename = 'index-static.bundle.js';
+  var indexStatic = null;
+  var buildInProgress = false;
+  var entry = path.join(__dirname, 'index-static.jsx');
+  var compiler = webpack({
+    entry: entry,
+    target: 'node',
+    devtool: 'sourcemap',
+    externals: function(context, request, callback) {
```

Comment:
Hey @toolness could you help me understand what `context` and `request` are?  I see they being mentioned on http://webpack.github.io/docs/configuration.html#externals but the doc doesn't go in detail.